sphinx/builders/html/__init__.py

```diff
@@ -718,6 +739,20 @@ def create_pygments_style_file(self) -> None:
         with open(path.join(self.outdir, '_static', 'pygments.css'), 'w') as f:
             f.write(self.highlighter.get_stylesheet())
 
+        if self.dark_highlighter:
+            with open(path.join(self.outdir, '_static', 'pygments_dark.css'), 'w') as f:
+                f.write(self.dark_highlighter.get_stylesheet())
+            self.add_css_file('pygments_dark.css',
+                              media='(prefers-color-scheme: dark)',
+                              id='pygments_dark_css')
```

Comment:
Is this working expected? I suppose this does not work fine because this method is called after writing HTML files. It would be better to call this in `init_highlighter()` instead.

Reply:
As far as I have tested this works as intended but adding them earlier may make it easier to overwrite these styles later so I will move this as you suggested